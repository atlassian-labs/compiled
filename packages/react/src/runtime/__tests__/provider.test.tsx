const mockNodeEnvFunction = jest.fn();

jest.mock('../is-node', () => ({
  isNodeEnvironment: mockNodeEnvFunction,
}));

const OLD_ENV = process.env;

describe('Provider index.tsx', () => {
  beforeEach(() => {
    jest.resetModules(); // most important - it clears the cache
    process.env = { ...OLD_ENV }; // make a copy
    process.env.NODE_ENV = 'development';
  });

  afterAll(() => {
    process.env = OLD_ENV; // restore old env
  });

  it('returns the browser provider when isNodeEnvironment returns false', () => {
    mockNodeEnvFunction.mockImplementation(() => false);

    const BrowserProvider = require('../provider/provider-browser').default;
    const Provider = require('../provider').default;

    expect(Provider).toEqual(BrowserProvider);
  });

  it('returns the server provider when isNodeEnvironment returns true', () => {
    mockNodeEnvFunction.mockImplementation(() => true);

    const ServerProvider = require('../provider/provider-server').default;
    const Provider = require('../provider').default;

    expect(Provider).toEqual(ServerProvider);
  });
});

describe('Server Provider', () => {
  beforeEach(() => {
    jest.resetModules(); // most important - it clears the cache
    process.env = { ...OLD_ENV }; // make a copy
    process.env.NODE_ENV = 'development';
  });

  afterAll(() => {
    process.env = OLD_ENV; // restore old env
  });

  it('should throw when "isNodeEnvironment" returns false', () => {
    mockNodeEnvFunction.mockImplementation(() => false);

    expect(() => require('../provider/provider-server')).toThrowErrorMatchingSnapshot();
  });

  it('should not throw when "isNodeEnvironment" returns true', () => {
    mockNodeEnvFunction.mockImplementation(() => true);

    expect(() => require('../provider/provider-server')).not.toThrowError();
  });
});

describe('Browser Provider', () => {
  beforeEach(() => {
    jest.resetModules(); // most important - it clears the cache
    process.env = { ...OLD_ENV }; // make a copy
    process.env.NODE_ENV = 'development';
  });

  afterAll(() => {
    process.env = OLD_ENV; // restore old env
  });

  it('should throw when "isNodeEnvironment" returns true', () => {
    mockNodeEnvFunction.mockImplementation(() => true);

    expect(() => require('../provider/provider-browser')).toThrowErrorMatchingSnapshot();
  });

  it('should not throw when "isNodeEnvironment" returns false', () => {
    mockNodeEnvFunction.mockImplementation(() => false);

    expect(() => require('../provider/provider-browser')).not.toThrowError();
  });
});
