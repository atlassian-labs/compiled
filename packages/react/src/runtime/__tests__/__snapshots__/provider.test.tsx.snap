// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browser Provider should throw when "isNodeEnvironment" returns true 1`] = `
"
 ██████╗ ██████╗ ███╗   ███╗██████╗ ██╗██╗     ███████╗██████╗
██╔════╝██╔═══██╗████╗ ████║██╔══██╗██║██║     ██╔════╝██╔══██╗
██║     ██║   ██║██╔████╔██║██████╔╝██║██║     █████╗  ██║  ██║
██║     ██║   ██║██║╚██╔╝██║██╔═══╝ ██║██║     ██╔══╝  ██║  ██║
╚██████╗╚██████╔╝██║ ╚═╝ ██║██║     ██║███████╗███████╗██████╔╝
 ╚═════╝ ╚═════╝ ╚═╝     ╚═╝╚═╝     ╚═╝╚══════╝╚══════╝╚═════╝

  @compiled/react/runtime - ERROR

  This code should only run on the client. You might need to configure your bundler to respect the \\"browser\\" field in package json.
"
`;

exports[`Server Provider should throw when "isNodeEnvironment" returns false 1`] = `
"
 ██████╗ ██████╗ ███╗   ███╗██████╗ ██╗██╗     ███████╗██████╗
██╔════╝██╔═══██╗████╗ ████║██╔══██╗██║██║     ██╔════╝██╔══██╗
██║     ██║   ██║██╔████╔██║██████╔╝██║██║     █████╗  ██║  ██║
██║     ██║   ██║██║╚██╔╝██║██╔═══╝ ██║██║     ██╔══╝  ██║  ██║
╚██████╗╚██████╔╝██║ ╚═╝ ██║██║     ██║███████╗███████╗██████╔╝
 ╚═════╝ ╚═════╝ ╚═╝     ╚═╝╚═╝     ╚═╝╚══════╝╚══════╝╚═════╝

  @compiled/react/runtime - ERROR

  This code should only run on the server. You might need to configure your bundler to respect the \\"browser\\" field in package json.
"
`;
